tests/integration_tests.rs

```diff
@@ -0,0 +1,776 @@
+
+    // In case of a seccomp denial `errno` should be `FAILURE_CODE`
+    if let Some(should_fail) = should_fail {
+        if should_fail {
+            assert_eq!(errno, FAILURE_CODE);
+        } else {
+            assert_ne!(errno, FAILURE_CODE);
+        }
+    }
+}
+
+#[test]
+fn test_empty_filter() {
+    // An empty filter should always return the default action.
+    // For example, for an empty allowlist, it should always trap/kill,
+    // for an empty denylist, it should allow allow all system calls.
+
+    let filter = SeccompFilter::new(
+        BTreeMap::new(),
+        SeccompAction::Allow,
+        SeccompAction::Trap,
+        ARCH.try_into().unwrap(),
+    )
+    .unwrap();
+    let prog: BpfProgram = filter.try_into().unwrap();
+
+    // This should allow any system calls.
+    let pid = thread::spawn(move || {
+        // Install the filter.
+        apply_filter(&prog).unwrap();
+
+        unsafe { libc::getpid() }
+    })
+    .join()
+    .unwrap();
+
+    // Check that the getpid syscall returned successfully.
+    assert!(pid > 0);
+}
+
+#[test]
+fn test_invalid_architecture() {
+    // A filter compiled for another architecture should kill the process upon evaluation.
+    // The process will appear as if it received a SIGSYS.
+    let mut arch = "aarch64";
+
+    if ARCH == "aarch64" {
+        arch = "x86_64";
+    }
+
+    let filter = SeccompFilter::new(
+        BTreeMap::new(),
+        SeccompAction::Allow,
+        SeccompAction::Trap,
+        arch.try_into().unwrap(),
+    )
+    .unwrap();
+    let prog: BpfProgram = filter.try_into().unwrap();
+
+    let pid = unsafe { libc::fork() };
+    match pid {
+        0 => {
+            apply_filter(&prog).unwrap();
+
+            unsafe {
+                libc::getpid();
+            }
+        }
+        child_pid => {
+            let mut child_status: i32 = -1;
+            let pid_done = unsafe { libc::waitpid(child_pid, &mut child_status, 0) };
+            assert_eq!(pid_done, child_pid);
+
+            assert!(libc::WIFSIGNALED(child_status));
+            assert_eq!(libc::WTERMSIG(child_status), libc::SIGSYS);
+        }
+    };
+}
+
+#[test]
+fn test_eq_operator() {
+    // check use cases for DWORD
+    let rules = vec![(
+        libc::SYS_ioctl,
+        vec![SeccompRule::new(vec![Cond::new(1, Dword, Eq, KVM_GET_PIT2).unwrap()]).unwrap()],
+    )];
+    // check syscalls that are supposed to work
+    validate_seccomp_filter(
+        rules.clone(),
+        || unsafe {
+            libc::ioctl(0, KVM_GET_PIT2 as IoctlRequest);
+        },
+        Some(false),
+    );
+    // check syscalls that are not supposed to work
+    validate_seccomp_filter(
+        rules,
+        || unsafe {
+            libc::ioctl(0, 0);
+        },
+        Some(true),
+    );
+
+    // check use cases for QWORD
+    let rules = vec![(
+        libc::SYS_ioctl,
+        vec![SeccompRule::new(vec![Cond::new(2, Qword, Eq, std::u64::MAX).unwrap()]).unwrap()],
```

Comment:
std::u64::MAX is deprecated. Can you replace it with `u64::MAX`?

Reply:
didn't know that, sure